### ISSUE TITLE
Terraform apply continues even when gcloud command fails

### ISSUE THREAD

**Issue by agutta** [terraform-google-modules/terraform-google-cloud-operations#64]
### TL;DR

Terraform apply is successful even when the gcloud command fails.

A few scenarios where terraform completes successfully but gcloud fails 
1) OS Config API not turned on
ERROR: (gcloud.beta.compute.instances.ops-agents.policies.create) User [user1@company.com] does not have permission to access projects instance [XYZ] (or it may not exist): OS Config API has not been used in project <Project_Number> before or it is disabled.  
2) Creating another policy with similar configuration as existing policy
(gcloud.beta.compute.instances.ops-agents.policies.create) INVALID_ARGUMENT: Conflict - Policy 'projects/<Project_Number>/guestPolicies/ops-agents-test' targets the same configs and instances.
3) Required fields missing
ERROR: (gcloud.beta.compute.instances.ops-agents.policies.update) argument --agent-rules: Key [enable-autoupgrade] required in dict arg but not provided

### Expected behavior

Above should give error message like below

ERROR: (gcloud.beta.compute.instances.ops-agents.policies.update) argument --agent-rules: Key [enable-autoupgrade] required in dict arg but not provided
Usage: gcloud beta compute instances ops-agents policies update POLICY_ID [optional flags]

### Observed behavior

Terraform shows  "Apply complete!"

### Terraform Configuration

```hcl
/*********************************************************
 OS Policies for OpsAgent Installation
*********************************************************/
locals {
  region = {
    us-central1_debian_11 = {
      "policy_id"   = "ops-agents-use1-debian-11"
      "description" = "us-central1 debian v11 policy"
      "zones"       = ["us-central1-a","us-central1-b","us-central1-c"]
      "short_name"  = "debian"
      "version"     = "11"
    }
  }
}

module "ops_agent_policy" {
  source     = "terraform-google-modules/cloud-operations/google//modules/agent-policy"
  version    = "~> 0.2.3"

  for_each    = local.region
  project_id  = var.project_id
  policy_id   = each.value.policy_id
  description = each.value.description
  zones       = each.value.zones
  agent_rules = [
    {
      type               = "ops-agent"
      version            = "current-major"
      # package-state      = "installed"
      # enable-autoupgrade = true
    },
  ]

  group_labels = [
    {
      grp        = "ops-agent"
    }
  ]

os_types = [
    {
      short_name = each.value.short_name
      version    = each.value.version
    },
  ]
}
```
```


### Terraform Version

```sh
Terraform v1.4.6
```

**Comment by lyricnz** [comment]
Yes, I see this a lot.
